### DISCUSSION TITLE
Windows Package Manager 1.6.2561

### DISCUSSION THREAD

**Discussion by ryfu-msft** [microsoft/winget-cli#3625] [Releases]
This release represents our second Windows Package Manager 1.6 release candidate build for Windows 10 (1809+), and Windows 11. Experimental features have been disabled in this release.

The [`winget configure`](https://learn.microsoft.com/windows/package-manager/configuration/) command is now a stable feature and can be used to automatically handle the setup and configuration requirements for an ideal development environment on your Windows machine. Applying a WinGet configuration file helps with installing and managing software packages, applications, programming languages, frameworks, tools, or settings necessary for a project.

You can also use the `winget download` command to download a package installer to your local machine. Support for package dependencies and enabling Windows Features are also included in this stable release.

## Features
* WinGet configure #2845
* Support for package dependencies #163 
* `winget download` #658 
* Specify InstallerType #1166 
* Windows Features (WSL) #3320 

## Bugs
* Symlinks are not created for portable installations #3498 

## What's Changed
* Bump version from 1.5 to 1.6 by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3317
* Create SUPPORT.md by @denelon in https://github.com/microsoft/winget-cli/pull/3340
* Update README.md by @denelon in https://github.com/microsoft/winget-cli/pull/3341
* Update README.md by @mdanish-kh in https://github.com/microsoft/winget-cli/pull/3342
* Fix for onboarding to GitOps.ResourceManagement by @Trenly in https://github.com/microsoft/winget-cli/pull/3350
* Onboarding to GitOps.ResourceManagement by @microsoft-github-policy-service in https://github.com/microsoft/winget-cli/pull/3347
* #2874: Fix for error git submodule status: fatal no submodule mapping… by @gigi81 in https://github.com/microsoft/winget-cli/pull/3305
* Move GitOps rules to their own files by @Trenly in https://github.com/microsoft/winget-cli/pull/3352
* Fix Component Governance alerts by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3355
* Add Breaking-Change label to comment triggers by @Trenly in https://github.com/microsoft/winget-cli/pull/3357
* Respect Group Policies for sources by @florelis in https://github.com/microsoft/winget-cli/pull/3367
* Stub upgrade self by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3299
* Refresh process path variable when installing package dependencies by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3296
* Network troubleshooting by @denelon in https://github.com/microsoft/winget-cli/pull/3389
* Microsoft.WinGet.Configuration samples by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3369
* Support for out of process configuration clients by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3363
* Force close sandbox server upon timeout by @yao-msft in https://github.com/microsoft/winget-cli/pull/3392
* Relax InstallationNotes max length by @yao-msft in https://github.com/microsoft/winget-cli/pull/3397
* Fix wingetutil nuget publish pipeline by @yao-msft in https://github.com/microsoft/winget-cli/pull/3396
* Improve packaged test log collection and fix crash by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3395
* Do not attempt post install ARP correlation if PackageFamilyName is provided and present for the user by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3391
* Support winget installing AppInstaller by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3377
* configure test command "implemented" by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3414
* Disable RTTI by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3422
* Explicitly close file stream on FileLogger destruction by @yao-msft in https://github.com/microsoft/winget-cli/pull/3424
* Add configuration binaries to binskim scan and fix issues by @yao-msft in https://github.com/microsoft/winget-cli/pull/3426
* Build fixes by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3433
* Fix using expired cert in tests by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3435
* Repair-WinGetPackageManager improvements by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3423
* Download command by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3376
* Add initial version of yaml manifest 1.6 by @yao-msft in https://github.com/microsoft/winget-cli/pull/3449
* Configure validate command by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3441
* Add missing definitions to release builds by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3450
* Populate missing ManifestVersion for manifest from rest source and make PackageFamilyName and installer type manifest validation warning by @yao-msft in https://github.com/microsoft/winget-cli/pull/3460
* Add file logger to the statics object creation by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3451
* Skip stub packages for msix installer validation by @yao-msft in https://github.com/microsoft/winget-cli/pull/3468
* Upgrade 1.6 schema to 2020-12 by @Trenly in https://github.com/microsoft/winget-cli/pull/3478
* Allow `--include-unknown` in list `--upgrade-available` by @florelis in https://github.com/microsoft/winget-cli/pull/3473
* Simplify creating local index by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3445
* Move functions to cmdlets for Microsoft.WinGet.Client by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3469
* Generate manifest for Winget Download by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3448
* Implement DownloadCommandProhibited by @yao-msft in https://github.com/microsoft/winget-cli/pull/3487
* Fix Component Governace issue with System.Security.Cryptography.Xml by @yao-msft in https://github.com/microsoft/winget-cli/pull/3495
* Revert "Down sampling (#2950)" by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3511
* Move Microsoft.WinGet.Client E2E test to Pester framework by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3503
* Don't copy processor's output binaries by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3526
* Attempt to prevent crash in `TelemetryTraceLogger::InitializeInternal()` by @florelis in https://github.com/microsoft/winget-cli/pull/3527
* PowerShellGet  by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3521
* Introducing new Group Policy EnableWindowsPackageManagerCommandLineInterfaces for Intune by @Madhusudhan-MSFT in https://github.com/microsoft/winget-cli/pull/3524
* Remove CrossProcessReaderWriteLock by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3549
* Guard WinRT InProc Window Package Manager Deployment APIs by EnableAp… by @Madhusudhan-MSFT in https://github.com/microsoft/winget-cli/pull/3537
* Replace download cancelled message with a localized message by @chausner in https://github.com/microsoft/winget-cli/pull/3559
* Add ability to specify where to save modules for configuration by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3536
* Make dependencies, windows feature, and download experimental features stable by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3545
* Rename file in single portable installs by @Trenly in https://github.com/microsoft/winget-cli/pull/3439
* Add UpgradeBehavior deny by @Trenly in https://github.com/microsoft/winget-cli/pull/3512
* Improve rest client manifest parsing for upgrade behavior deny by @yao-msft in https://github.com/microsoft/winget-cli/pull/3570
* Add support for `--Installer-Type` argument for commands by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3516
* Enable configuration via IConfigurationStatics by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3576
* Apply loc update patch by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3594
* Give System and Admins full access to state folders by @yao-msft in https://github.com/microsoft/winget-cli/pull/3471
* Update configuration interface for future design changes by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3584
* Make config not experimental and add GP to control it by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3585
* make windows feature experimental by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3620

## New Contributors
* @microsoft-github-policy-service made their first contribution in https://github.com/microsoft/winget-cli/pull/3347
* @gigi81 made their first contribution in https://github.com/microsoft/winget-cli/pull/3305

**Full Changelog**: https://github.com/microsoft/winget-cli/compare/v1.5.2201...v1.6.2561

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/winget-cli/releases/tag/v1.6.2561'>Windows Package Manager 1.6.2561</a>.</em>

**Comment by Glem29** [comment]
Hi,
We are facing the problem of access for Admins or System to state folders. https://github.com/microsoft/winget-cli/pull/3471
So, we can't delete user profiles with our processes anymore..
I tried to update my Windows Package Manager 1.5.2201 with this release.
Unfortunately, rights doesn't seem to be given for System or Admins to the Default State Folder.
Is this release committed and downloadable yet ?
Thanks

**Reply by denelon** [reply]
@Glem29 I believe the fix may be in the 1.6 release candidate build linked above.

**Reply by Glem29** [reply]
I don't know if i'm at the right place to talk about that. Please give me an advice for rerouting if you have an idea.
About the https://github.com/microsoft/winget-cli/pull/3471
When i succeed in having good rights on the "statefolder" directory, some updates are ressetting these rights. 
I think the more eligible process is the Company Portal App which reset these rights....( seen the 20thSeptember)
So, my question is : If Windows Package Installer fixes the "rights" problem, will it be sufficient in all the Microsoft environment

**Reply by denelon** [reply]
I don't have full context on what the Company Portal App is doing with respect to permissions. Since WinGet is not running as a service, the only mechanism it has to "inspect" or "modify" rights is only triggered when it's executed.
